### ISSUE TITLE
There is no way to refresh the classpath after adding a library to gradle

### ISSUE THREAD

**Issue by zSynx94** [redhat-developer/vscode-java#3893]
I added a new library to my java project (org.lz4:lz4-java:1.8.0) but there is no way to get them indexed again.
I tried _gradlew build_ and _gradle --refresh-dependencies_ but it din't work.

**Environment**
Operating System: Windows 10 Enterprise LTSC 21H2
Visual Studio Code version: 1.85.2
Language Support for Java(TM) by Red Hat Version: 1.37.0
JDK: Amazon Corretto jdk1.8.0_432
Gradle Version: 4.7

**Steps To Reproduce**
Update/add dependencies to build.gradle file
Wait for it to update (nothing will happen)

**Captures**
![Image](https://github.com/user-attachments/assets/ba4e4192-97e2-41d6-86d5-306a3d149fe7)
![Image](https://github.com/user-attachments/assets/f00eb1be-8110-4a37-a109-4e17873d3641)
![Image](https://github.com/user-attachments/assets/9bdde6aa-5d95-4a9a-9ec7-d46c34432e4f)

**Comment by snjeza** [comment]
@zSynx94 I can't reproduce the issue. Could you attach a project example reproducing the error?

**Comment by linuxhelf** [comment]
i am having the same problem.
Neither the Task "Java: Reload Projects" nor the popup, that appears when changing the build.gradle, which wants to synchronize the classpath works for me.
Only "Java: Clean Java Language Server Workspace", which gets anoying in larger projects.

@snjeza 
I have created an example project at https://github.com/linuxhelf/vscode-java-import
Steps to reproduce:

- Open project and wait for it to build
- see error in App.java
- uncomment the org.json import in app/build.gradle
- error in App.java is still present and cant be fixed by reload

What i found out while creating the example project:
**This problem is only present with "java.gradle.buildServer.enabled": "off"**
When using the gradle buildserver, both the popup and "Java: Reload Projects" correctly update the classpath.

@zSynx94 are you also using "java.gradle.buildServer.enabled": "off"?

**Comment by rgrunber** [comment]
@snjeza , are you and others able to confirm the problem also goes away when setting `java.configuration.updateBuildConfiguration": "automatic"` and restarting VS Code ? (restart is needed because there's another bug where it seems changing the setting isn't respected until VS Code is reloaded). Overall, I noticed I could only reproduce the problem when the setting is set to `interactive` (popup comes up to ask if you'd like to update, given that a build file was changed) and without the gradle build server (so this is using Buildship).

**Comment by linuxhelf** [comment]
@rgrunber seems complicated.... i tried to reproduce your case and it didn't work... and then suddenly it worked 

i tried a bit more, if i understand it correctly, it behaves as following

first case (starting with automatic):

- start with clean workspace & java.configuration.updateBuildConfiguration": "automatic"
- adding / removing the dependency from build.gradle correctly adds/removes it from the classpath
- setting to  java.configuration.updateBuildConfiguration": "interactive"
- adding / removing the dependency from build.gradle correctly adds/removes it from the classpath without clickling the popup

second case (starting with interactive)

- start with clean workspace & java.configuration.updateBuildConfiguration": "interactive"
- adding / removing the dependency from build.gradle does not adds/removes it from the classpath
- setting to  java.configuration.updateBuildConfiguration": "automatic"
- adding / removing the dependency from build.gradle does not adds/removes it from the classpath


both times i also tried closing/reopening VS Code, but that did noch change anything for me

so conclusion:
this issue only exists with 
`"java.gradle.buildServer.enabled": "off"` and `java.configuration.updateBuildConfiguration": "interactive"`
and for `java.configuration.updateBuildConfiguration"` it takes the value when the workspace was first initialised

**Comment by snjeza** [comment]
>  are you and others able to confirm the problem also goes away when setting java.configuration.updateBuildConfiguration": "automatic" and restarting VS Code ? (restart is needed because there's another bug where it seems changing the setting isn't respected until VS Code is reloaded). Overall, I noticed I could only reproduce the problem when the setting is set to interactive (popup comes up to ask if you'd like to update, given that a build file was changed) and without the gradle build server (so this is using Buildship).

@rgrunber Please see https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/3349#issuecomment-2573781815

**Comment by snjeza** [comment]
@linuxhelf Could you try to set
```
auto.sync=true
```
to `vscode-java-import/app/.settings/org.eclipse.buildship.core.prefs`?

**Comment by linuxhelf** [comment]
@snjeza auto.sync=true had no impact for me, only java.configuration.updateBuildConfiguration made a difference

i tried both `java.configuration.updateBuildConfiguration` `automatic` and `interactive `with both `auto.sync` `true` and `false`
and i used Clean Java Language Server Workspace at the start for each combination

**Comment by rgrunber** [comment]
@snjeza I"m seeing the exact behaviour described in https://github.com/redhat-developer/vscode-java/issues/3893#issuecomment-2573709105. Looking into it, it seems to me like `auto.sync` is enabled when the project is imported with `"java.configuration.updateBuildConfiguration": "automatic"` being the initial setting and disabled when `"java.configuration.updateBuildConfiguration": "interactive"` is the initial setting on import.

**Comment by snjeza** [comment]
>  and disabled when "java.configuration.updateBuildConfiguration": "interactive" is the initial setting on import.

`auto.sync=false` is set by buildship by default.

**Comment by snjeza** [comment]
> That probably would explain why changing from interactive to automatic doesn't work, and why going from automatic to interactive updates without waiting for the prompt.

Right. When `auto.sync=true`, buildship updates the project without waiting for the prompt. This PR doesn't introduce it.
You can create a new issue if it is a problem.

**Comment by rgrunber** [comment]
Ok, then I'll evaluate your PR while ignoring that bug. I'd also look back at old issues and see if this has ever been filed before. I'm surprised such an issue would have existed for so long, especially given `interactive` is the default.

**Comment by snjeza** [comment]
> Ok, then I'll evaluate your PR while ignoring that bug. I'd also look back at old issues and see if this has ever been filed before. I'm surprised such an issue would have existed for so long, especially given interactive is the default.

The related PR - https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2224
